app/models/task.py

```diff
@@ -1,31 +1,36 @@
-from sqlalchemy.orm import Mapped, mapped_column, relationship
-from ..db import db
 from datetime import datetime
 from typing import Optional
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from ..db import db
 from .goal import Goal
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
     title: Mapped[str] = mapped_column()
     description: Mapped[str] = mapped_column()
-    completed_at: Mapped[datetime] = mapped_column(nullable=True)
-    goal_id: Mapped[Optional[int]] = mapped_column(db.ForeignKey("goal.id"), nullable=True)
+    completed_at: Mapped[Optional[datetime]] = mapped_column()
+    goal_id: Mapped[Optional[int]] = mapped_column(db.ForeignKey("goal.id"))
     goal: Mapped[Optional["Goal"]] = relationship(back_populates="tasks")
 
-    def to_dict(self, include_goal_id=False):
+    def is_complete(self):
+        return self.completed_at is not None
+
+    def slack_completion_message(self):
+        return f"Jamie just completed the task: {self.title}"
```

Comment:
I'd actually avoid adding this directly to the Model. The model doesn't necessarily need to know about how to create completion messages. Notice that we added the Slack messaging as a step separate from the actual completion (we authored the endpoints in wave 3, then added the notification in wave 4). The completion itself (updating the completed_at value) would be a good candidate for a Model helper, but I would move this logic to a separate class/function related to our Slack subsystem. We could pass in the the task, or the title and it would build the message.